src/librustc_codegen_llvm/back/lto.rs

```diff
@@ -453,12 +515,36 @@ fn thin_lto(diag_handler: &Handler,
             serialized_modules: serialized,
             module_names,
         });
-        Ok((0..shared.module_names.len()).map(|i| {
-            LtoModuleCodegen::Thin(ThinModule {
+
+        let mut copy_jobs = vec![];
+        let mut opt_jobs = vec![];
+
+        for (module_index, module_name) in shared.module_names.iter().enumerate() {
+            let module_name = module_name_to_str(module_name);
+
+            if green_modules.contains_key(module_name) {
+                let mut imports_all_green = true;
```

Comment:
This might be a bit clearer as:

```rust
let imports_all_green = import_map.modules_imported_by(module_name)
    .all(|imported_module| green_module.contains_key(imported_module));
```